File: frontend/admin/packages/modules/system/src/api/system/admin/v1/base_oauth_provider.ts
import service from "@liujitcn/kratos-admin-core/request";
import type {
  BaseOauthProviderForm,
  BaseOauthProviderService,
  CreateBaseOauthProviderRequest,
  DeleteBaseOauthProviderRequest,
  GetBaseOauthProviderRequest,
  PageBaseOauthProviderRequest,
  PageBaseOauthProviderResponse,
  SetBaseOauthProviderStatusRequest,
  UpdateBaseOauthProviderRequest
} from "@liujitcn/kratos-admin-system/rpc/system/admin/v1/base_oauth_provider";
import type { Empty } from "@liujitcn/kratos-admin-system/rpc/google/protobuf/empty";

const OAUTH_PROVIDER_URL = "/v1/admin/base/oauth-provider";

/** OAuth第三方登录方式管理服务实现。 */
export class BaseOauthProviderServiceImpl implements BaseOauthProviderService {
  /** 分页查询OAuth登录方式。 */
  PageBaseOauthProvider(request: PageBaseOauthProviderRequest): Promise<PageBaseOauthProviderResponse> {
    return service({ url: OAUTH_PROVIDER_URL, method: "get", params: request });
  }

  /** 查询OAuth登录方式详情。 */
  GetBaseOauthProvider(request: GetBaseOauthProviderRequest): Promise<BaseOauthProviderForm> {
    return service({ url: `${OAUTH_PROVIDER_URL}/${request.id}`, method: "get" });
  }

  /** 创建OAuth登录方式。 */
  CreateBaseOauthProvider(request: CreateBaseOauthProviderRequest): Promise<Empty> {
    return service({ url: OAUTH_PROVIDER_URL, method: "post", data: request.base_oauth_provider });
  }

  /** 更新OAuth登录方式。 */
  UpdateBaseOauthProvider(request: UpdateBaseOauthProviderRequest): Promise<Empty> {
    return service({ url: `${OAUTH_PROVIDER_URL}/${request.base_oauth_provider?.id ?? ""}`, method: "put", data: request.base_oauth_provider });
  }

  /** 删除OAuth登录方式。 */
  DeleteBaseOauthProvider(request: DeleteBaseOauthProviderRequest): Promise<Empty> {
    return service({ url: `${OAUTH_PROVIDER_URL}/${request.id}`, method: "delete" });
  }

  /** 设置OAuth登录方式状态。 */
  SetBaseOauthProviderStatus(request: SetBaseOauthProviderStatusRequest): Promise<Empty> {
    return service({ url: `${OAUTH_PROVIDER_URL}/${request.id}/status`, method: "put", data: request });
  }
}

export const defBaseOauthProviderService = new BaseOauthProviderServiceImpl();

File: backend/internal/service/system/admin/v1/base_oauth_provider.go
package admin

import (
	"context"
	"fmt"

	adminv1 "github.com/liujitcn/kratos-admin/backend/api/gen/go/system/admin/v1"
	biz "github.com/liujitcn/kratos-admin/backend/internal/biz/system/admin"
	"github.com/liujitcn/kratos-core/errorsx"

	"github.com/go-kratos/kratos/v3/log"
	"google.golang.org/protobuf/types/known/emptypb"
)

// BaseOauthProviderService OAuth第三方登录方式管理服务。
type BaseOauthProviderService struct {
	adminv1.UnimplementedBaseOauthProviderServiceServer
	baseOauthProviderCase *biz.BaseOauthProviderCase
}

// NewBaseOauthProviderService 创建 OAuth 第三方登录方式管理服务。
func NewBaseOauthProviderService(baseOauthProviderCase *biz.BaseOauthProviderCase) *BaseOauthProviderService {
	return &BaseOauthProviderService{baseOauthProviderCase: baseOauthProviderCase}
}

// PageBaseOauthProvider 分页查询 OAuth 登录方式。
func (s *BaseOauthProviderService) PageBaseOauthProvider(ctx context.Context, req *adminv1.PageBaseOauthProviderRequest) (*adminv1.PageBaseOauthProviderResponse, error) {
	result, err := s.baseOauthProviderCase.PageBaseOauthProvider(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("PageBaseOauthProvider %v", err))
		return nil, errorsx.WrapInternal(err, "查询OAuth登录方式失败")
	}
	return result, nil
}

// GetBaseOauthProvider 查询 OAuth 登录方式详情。
func (s *BaseOauthProviderService) GetBaseOauthProvider(ctx context.Context, req *adminv1.GetBaseOauthProviderRequest) (*adminv1.BaseOauthProviderForm, error) {
	result, err := s.baseOauthProviderCase.GetBaseOauthProvider(ctx, req.GetId())
	if err != nil {
		log.Error(fmt.Sprintf("GetBaseOauthProvider %v", err))
		return nil, errorsx.WrapInternal(err, "查询OAuth登录方式失败")
	}
	return result, nil
}

// CreateBaseOauthProvider 创建 OAuth 登录方式。
func (s *BaseOauthProviderService) CreateBaseOauthProvider(ctx context.Context, req *adminv1.CreateBaseOauthProviderRequest) (*emptypb.Empty, error) {
	err := s.baseOauthProviderCase.CreateBaseOauthProvider(ctx, req.GetBaseOauthProvider())
	if err != nil {
		log.Error(fmt.Sprintf("CreateBaseOauthProvider %v", err))
		return nil, errorsx.WrapInternal(err, "创建OAuth登录方式失败")
	}
	return new(emptypb.Empty), nil
}

// UpdateBaseOauthProvider 更新 OAuth 登录方式。
func (s *BaseOauthProviderService) UpdateBaseOauthProvider(ctx context.Context, req *adminv1.UpdateBaseOauthProviderRequest) (*emptypb.Empty, error) {
	err := s.baseOauthProviderCase.UpdateBaseOauthProvider(ctx, req.GetBaseOauthProvider())
	if err != nil {
		log.Error(fmt.Sprintf("UpdateBaseOauthProvider %v", err))
		return nil, errorsx.WrapInternal(err, "更新OAuth登录方式失败")
	}
	return new(emptypb.Empty), nil
}

// DeleteBaseOauthProvider 删除 OAuth 登录方式。
func (s *BaseOauthProviderService) DeleteBaseOauthProvider(ctx context.Context, req *adminv1.DeleteBaseOauthProviderRequest) (*emptypb.Empty, error) {
	err := s.baseOauthProviderCase.DeleteBaseOauthProvider(ctx, req.GetId())
	if err != nil {
		log.Error(fmt.Sprintf("DeleteBaseOauthProvider %v", err))
		return nil, errorsx.WrapInternal(err, "删除OAuth登录方式失败")
	}
	return new(emptypb.Empty), nil
}

// SetBaseOauthProviderStatus 设置 OAuth 登录方式状态。
func (s *BaseOauthProviderService) SetBaseOauthProviderStatus(ctx context.Context, req *adminv1.SetBaseOauthProviderStatusRequest) (*emptypb.Empty, error) {
	err := s.baseOauthProviderCase.SetBaseOauthProviderStatus(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("SetBaseOauthProviderStatus %v", err))
		return nil, errorsx.WrapInternal(err, "设置OAuth登录方式状态失败")
	}
	return new(emptypb.Empty), nil
}
